From 6c6d713927bfe19627fc22ec67a211d6a1a28fc9 Mon Sep 17 00:00:00 2001
Message-Id: <6c6d713927bfe19627fc22ec67a211d6a1a28fc9.1411422080.git.crobinso@redhat.com>
From: Cole Robinson <crobinso@redhat.com>
Date: Sun, 21 Sep 2014 15:41:51 -0400
Subject: [PATCH libosinfo] oses: Add Fedora21

This just adds a plain Fedora21 entry. F21 is going to have different
versions like 'Workstation' and 'Server' but we don't seem to distinguish
for other distros so I just stuck with the pattern here.

This currently doesn't work for detecting Workstation media, since their
volume IDs are truncated. Bug tracking that:

https://bugzilla.redhat.com/show_bug.cgi?id=1145264
Signed-off-by: Cole Robinson <crobinso@redhat.com>
---
 data/oses/fedora.xml.in                            | 86 ++++++++++++++++++++++
 .../Fedora-Server-DVD-x86_64-21_Alpha_TC7.iso.txt  | 29 ++++++++
 2 files changed, 115 insertions(+)
 create mode 100644 test/isodata/fedora/fedora21/Fedora-Server-DVD-x86_64-21_Alpha_TC7.iso.txt

diff --git a/data/oses/fedora.xml.in b/data/oses/fedora.xml.in
index 27cdd4a..1848a78 100644
--- a/data/oses/fedora.xml.in
+++ b/data/oses/fedora.xml.in
@@ -1584,4 +1584,90 @@
       </minimum>
     </resources>
   </os>
+
+  <os id="http://fedoraproject.org/fedora/21">
+    <short-id>fedora21</short-id>
+    <_name>Fedora 21</_name>
+    <version>21</version>
+    <_vendor>Fedora Project</_vendor>
+    <family>linux</family>
+    <distro>fedora</distro>
+    <upgrades id="http://fedoraproject.org/fedora/20"/>
+    <derives-from id="http://fedoraproject.org/fedora/20"/>
+
+    <media arch="i686">
+      <!-- <url>http://download.fedoraproject.org/pub/fedora/linux/releases/21/
+Fedora/i386/iso/Fedora-21-i386-DVD.iso</url> -->
+      <iso>
+        <volume-id>Fedora.*21.*-i386</volume-id>
+        <system-id>LINUX</system-id>
+      </iso>
+      <kernel>isolinux/vmlinuz</kernel>
+      <initrd>isolinux/initrd.img</initrd>
+    </media>
+    <media arch="x86_64">
+      <!-- <url>http://download.fedoraproject.org/pub/fedora/linux/releases/21/
+Fedora/x86_64/iso/Fedora-21-x86_64-DVD.iso</url> -->
+      <iso>
+        <volume-id>Fedora.*-21.*-x86_64</volume-id>
+        <system-id>LINUX</system-id>
+      </iso>
+      <kernel>isolinux/vmlinuz</kernel>
+      <initrd>isolinux/initrd.img</initrd>
+    </media>
+
+    <!-- Workstation live media has a truncated volume ID at the moment
+         which doesn't include the actual Fedora version, so this won't work
+         https://bugzilla.redhat.com/show_bug.cgi?id=1145264 -->
+    <media arch="i686" live="true">
+      <!-- <url>http://download.fedoraproject.org/pub/fedora/linux/releases/21/Live/i386/Fedora-Live-Desktop-i686-21-1.iso</url> -->
+      <iso>
+        <volume-id>Fedora-Live-.*-i686-21</volume-id>
+        <system-id>LINUX</system-id>
+      </iso>
+      <kernel>isolinux/vmlinuz0</kernel>
+      <initrd>isolinux/initrd0.img</initrd>
+    </media>
+    <media arch="x86_64" live="true">
+      <!-- <url>http://download.fedoraproject.org/pub/fedora/linux/releases/21/Live/x86_64/Fedora-Live-Desktop-x86_64-21-1.iso</url> -->
+      <iso>
+        <volume-id>Fedora-Live-.*-x86_64-21</volume-id>
+        <system-id>LINUX</system-id>
+      </iso>
+      <kernel>isolinux/vmlinuz0</kernel>
+      <initrd>isolinux/initrd0.img</initrd>
+    </media>
+
+    <tree arch="i686">
+      <!-- <url>http://download.fedoraproject.org/pub/fedora/linux/releases/21/Fedora/i386/os/</url> -->
+      <treeinfo>
+        <family>Fedora</family>
+        <version>21</version>
+        <arch>i386</arch>
+      </treeinfo>
+    </tree>
+    <tree arch="x86_64">
+      <!-- <url>http://download.fedoraproject.org/pub/fedora/linux/releases/21/Fedora/x86_64/os/</url> -->
+      <treeinfo>
+        <family>Fedora</family>
+        <version>21</version>
+        <arch>x86_64</arch>
+      </treeinfo>
+    </tree>
+
+    <resources arch="all">
+      <minimum>
+        <n-cpus>1</n-cpus>
+        <cpu>1000000000</cpu>
+        <ram>1073741824</ram>
+        <storage>10737418240</storage>
+      </minimum>
+    </resources>
+
+    <!-- Lets uncomment these after testing them -->
+    <!--<installer>
+      <script id='http://fedoraproject.org/scripts/fedora/jeos'/>
+      <script id='http://fedoraproject.org/scripts/fedora/desktop'/>
+    </installer>-->
+  </os>
 </libosinfo>
diff --git a/test/isodata/fedora/fedora21/Fedora-Server-DVD-x86_64-21_Alpha_TC7.iso.txt b/test/isodata/fedora/fedora21/Fedora-Server-DVD-x86_64-21_Alpha_TC7.iso.txt
new file mode 100644
index 0000000..f4e9436
--- /dev/null
+++ b/test/isodata/fedora/fedora21/Fedora-Server-DVD-x86_64-21_Alpha_TC7.iso.txt
@@ -0,0 +1,29 @@
+CD-ROM is in ISO 9660 format
+System id: LINUX
+Volume id: Fedora-S-21_A_T7-x86_64
+Volume set id: 
+Publisher id: 
+Data preparer id: 
+Application id: GENISOIMAGE ISO 9660/HFS FILESYSTEM CREATOR (C) 1993 E.YOUNGDALE (C) 1997-2006 J.PEARSON/J.SCHILLING (C) 2006-2007 CDRKIT TEAM
+Copyright File id: 
+Abstract File id: 
+Bibliographic File id: 
+Volume set size is: 1
+Volume set sequence number is: 1
+Logical block size is: 2048
+Volume size is: 979923
+El Torito VD version 1 found, boot catalog is in sector 439
+Joliet with UCS level 3 found
+Rock Ridge signatures version 1 found
+Eltorito validation header:
+    Hid 1
+    Arch 0 (x86)
+    ID ''
+    Key 55 AA
+    Eltorito defaultboot header:
+        Bootid 88 (bootable)
+        Boot media 0 (No Emulation Boot)
+        Load segment 0
+        Sys type 0
+        Nsect 4
+        Bootoff 1B8 440
-- 
2.1.0

